### ISSUE TITLE
Git warning of a missing default when upgrading Flutter

### ISSUE THREAD

**Issue by timsneath** [flutter/flutter#54983]
On running `flutter upgrade` on `master`, this error is now shown:

```
$ flutter upgrade
Upgrading Flutter from /Users/timsneath/flutter...
warning: Pulling without specifying how to reconcile divergent branches is
discouraged. You can squelch this message by running one of the following
commands sometime before your next pull:

  git config pull.rebase false  # merge (the default strategy)
  git config pull.rebase true   # rebase
  git config pull.ff only       # fast-forward only

You can replace "git config" with "git config --global" to set a default
preference for all repositories. You can also pass --rebase, --no-rebase,
or --ff-only on the command line to override the configured default per
invocation.
```

This is a [new warning from Git](https://public-inbox.org/git/20200304022931.2469455-1-alexhenrie24@gmail.com/), but it's likely confusing to casual Flutter users, who will not expect to have to deal with this or necessarily know which is the right option to select for Flutter. 

As the warning notes, you can "pass `--rebase`, `--no-rebase`, or `--ff-only` on the command line to override the configured default per invocation" on those overrides when we call `git` from `flutter upgrade`. We should do that, since it doesn't disturb the user's config.

**Comment by timsneath** [comment]
Boom! How about that for a fast turnaround! Thanks, @jmagman!

**Comment by pcsosinski** [comment]
does this affect 1.17 as well?

**Comment by timsneath** [comment]
If I've understood the underlying problem, this is orthogonal to Flutter release -- anyone who updates to the latest version of Git will start hitting it until they upgrade to a version that includes @jmagman's commit. So getting this out in 1.17 is valuable. Please correct me if I'm wrong, @jmagman or @christopherfujino!

**Comment by christopherfujino** [comment]
> If I've understood the underlying problem, this is orthogonal to Flutter release -- anyone who updates to the latest version of Git will start hitting it until they upgrade to a version that includes @jmagman's commit. So getting this out in 1.17 is valuable. Please correct me if I'm wrong, @jmagman or @christopherfujino!

Correct, this will affect all versions of flutter that don't have this change.

**Comment by pcsosinski** [comment]
fixed in 1.17.0-3.2.pre
